website/server/controllers/api-v3/auth.js

```diff
@@ -239,17 +241,39 @@ function _passportFbProfile (accessToken) {
   });
 }
 
+function _passportGoogleProfile (accessToken) {
+  return new Bluebird((resolve, reject) => {
+    passport._strategies.google.userProfile(accessToken, (err, profile) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(profile);
+      }
+    });
+  });
+}
+
+function _passportProfile (network, accessToken) {
+  if (network === 'facebook') {
```

Comment:
You could also attach these functions to an object like:

```js
let passportProfiles = {
  facebook: _passportFbProfile,
  google: _passportGoogleProfile,
}

function _passportProfile (network, accessToken) {
  if (network in passportProfiles) {
    return passportProfiles[network](accessToken);
  }
}
```

Then adding additional logins is as trivial as updating the object mapping.